indi-qhy/qhy_focuser.h

```diff
@@ -29,47 +33,57 @@ class QFocuser : public INDI::Focuser
     virtual void TimerHit() override;
 
 private:
-  // Connection::Serial *serialConnection{nullptr};
   int SendCommand(char *cmd_line);
   int ReadResponse(char *buf);
   void GetFocusParams();
 
   int updatePosition(double *value);
   int updateTemperature(double *value);
-  int updateBacklash(double *value);
-  int updateMotorSettings(double *duty, double *delay, double *ticks);
   int updatePositionRelativeInward(double value);
   int updatePositionRelativeOutward(double value);
   int updatePositionAbsolute(double value);
 
-  int updateMaxPosition(double *value);
   int updateSetPosition(int value);
 
-  int updateSetSpeed(int value);
+  int updateSetReverse(int value);
 
-  int ReadUntilComplete(char *buf, int timeout);
+  int updateSetSpeed(int value);
 
   int timerID { -1 };
   bool initTargetPos = true;
   double targetPos{ 0 };
   double simulatedTemperature{ 0 };
   double simulatedPosition{ 0 };
+  bool isReboot = false;
+  int RebootTimes = 0;
 
   uint8_t buff[USB_CDC_RX_LEN];
 
+  int32_t cmd_version;
+  int32_t cmd_version_board;
   int32_t cmd_position;
   int32_t cmd_out_temp;
   int32_t cmd_chip_temp;
+  int32_t cmd_voltage;
+
 
   ISwitch ResetPosS[1];
   ISwitchVectorProperty ResetPosSP;
-  ISwitch RevertDirS[2];
-  ISwitchVectorProperty RevertDirSP;
+  // ISwitch RevertDirS[2];
+  // ISwitchVectorProperty RevertDirSP;
 
   INumber TemperatureChip[1];
   INumberVectorProperty TemperatureChipVP;
   INumber TemperatureN[1];
   INumberVectorProperty TemperatureNP;
+  INumber Voltage[1];
+  INumberVectorProperty VoltageVP;
+
+  INumber FOCUSVersion[1];
+  INumberVectorProperty FOCUSVersionVP;
+
+  INumber BOARDVersion[1];
+  INumberVectorProperty BOARDVersionVP;
```

Comment:
Please use INDI::PropertyNumber instead as INumber/INumberVectorProperty is deprecated for new drivers.